dev/tests/integration/testsuite/Magento/Catalog/Helper/Product/ViewTest.php

```diff
@@ -50,7 +50,7 @@ protected function setUp()
         ];
         $context = $this->objectManager->create('Magento\Framework\App\Action\Context', $arguments);
         $this->_controller = $this->objectManager->create(
-            'Magento\Catalog\Controller\Product',
+            'Magento\Catalog\Helper\Product\Stub\ProductControllerStub',
```

Comment:
Didn't want to use a real class because that would potentially change the test behavior. Wanted to keep it as close to the current state as possible.
I didn't use `getMockForAbstractClass` since I wanted to try to stick with real code as much as possible within the integration tests. A simple stub implementation seemed like the least intrusive way to go.
I don't really care though. If you prefer a mock of the abstract instead, no prob.